x-pack/legacy/plugins/apm/public/components/app/Settings/AddSettings/AddSettingFlyout.tsx

```diff
@@ -143,6 +148,7 @@ export function AddSettingsFlyout({
             serviceName={serviceName}
             setServiceName={setServiceName}
             sampleRate={sampleRate}
+            sampleRateFloat={sampleRateFloat}
```

Comment:
Is `sampleRateFloat` really needed in `AddSettingFlyoutBody`? I can see we use it for some validation but `isSampleRateValid` is already passed as an argument. Why is that not sufficient?

Reply:
I think if possible `AddSettingFlyoutBody` should not have any of this validation logic. Instead it should just receive the values it needs and matching booleans for whether they are valid.

Reply:
Or they could be grouped in pairs:

```js
const sampleRate = {
  value: '12.43',
  isValid: true
}
```